integrations/astra/tests/test_retriever.py

```diff
@@ -3,10 +3,36 @@
 # SPDX-License-Identifier: Apache-2.0
 from unittest.mock import patch
 
+import pytest
+from haystack.document_stores.types import FilterPolicy
+
 from haystack_integrations.components.retrievers.astra import AstraEmbeddingRetriever
 from haystack_integrations.document_stores.astra import AstraDocumentStore
 
 
+@patch.dict(
+    "os.environ",
+    {"ASTRA_DB_APPLICATION_TOKEN": "fake-token", "ASTRA_DB_API_ENDPOINT": "http://fake-url.apps.astra.datastax.com"},
+)
+@patch("haystack_integrations.document_stores.astra.document_store.AstraClient")
+def test_retriever_init(*_):
+    ds = AstraDocumentStore()
+    retriever = AstraEmbeddingRetriever(ds, filters={"foo": "bar"}, top_k=99, filter_policy="replace")
+    assert retriever.filters == {"foo": "bar"}
+    assert retriever.top_k == 99
+    assert retriever.document_store == ds
+    assert retriever.filter_policy == FilterPolicy.REPLACE
+
+    retriever = AstraEmbeddingRetriever(ds, filters={"foo": "bar"}, top_k=99, filter_policy=FilterPolicy.MERGE)
+    assert retriever.filter_policy == FilterPolicy.MERGE
+
+    with pytest.raises(ValueError, match="Unknown FilterPolicy"):
```

Comment:
```suggestion
    with pytest.raises(ValueError):
```

I think we prefer not matching error messages.